File: core/src/main/java/org/elkd/core/server/cluster/ClusterSet.java
package org.elkd.core.server.cluster;

import java.util.Set;

/**
 * Cluster membership container.
 */
public interface ClusterSet {
  int size();

  /**
   * Reserved for dynamic cluster configuration.
   *
   * @param node addKV the given target to the cluster.
   */
  void addNode(Node node);

  /**
   * Reserved for dynamic cluster configuration.
   *
   * @param node removed the given target from the cluster.
   */
  void removeNode(Node node);

  Set<Node> getNodes();

  Set<Node> getAllNodes();

  Node getLocalNode();

  boolean isEmpty();
}

File: core/src/main/java/org/ravine/core/server/cluster/ClusterSet.kt
package org.ravine.core.server.cluster

/**
 * Cluster membership container.
 */
interface ClusterSet {

  val nodes: Set<Node>

  val allNodes: Set<Node>

  val selfNode: Node

  val isEmpty: Boolean

  /**
   * Returns size of cluster.
   */
  fun size(): Int

  /**
   * Reserved for dynamic cluster configuration.
   *
   * @param node addKV the given target to the cluster.
   */
  fun addNode(node: Node)

  /**
   * Reserved for dynamic cluster configuration.
   *
   * @param node removed the given target from the cluster.
   */
  fun removeNode(node: Node)
}
